editor/plugins/shader_editor_plugin.cpp

```diff
@@ -196,6 +196,57 @@ VisualShaderEditor *ShaderEditorPlugin::get_visual_shader_editor(const Ref<Shade
 	return nullptr;
 }
 
+void ShaderEditorPlugin::set_window_layout(Ref<ConfigFile> p_layout) {
+	if (!p_layout->has_section("ShaderEditor")) {
```

Comment:
This should align with `script_editor` and check `EDITOR_GET("text_editor/behavior/files/restore_scripts_on_load")`

Reply:
I'm not sure about this. It might be misleading for users that this setting also affects the shader editor, in which you can edit both textual and visual shaders. Maybe we should provide a separate editor flag?

Reply:
I guess it depends, I would class shaders as scripts. Either a new or sharing that setting is fine with me.